packages/frontend/src/components/actionDetail.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react'
+import { InfoType } from '../constants'
+import HelpWidget from './helpWidget'
+import { shortenEpochKey } from '../utils'
+
+type Props = {
+    showBorder: boolean
+    showHelp: boolean
+    showRep: boolean
+    maxRep: number
+    defaultRep: number
+    hasRep: number
+    showoffRep: number
+    setShowoffRep: (showoffRep: number) => void
+    allEpks: string[]
+    useSubsidy: boolean
+    chooseToUseSubsidy: () => void
+    chooseToUsePersona: () => void
+    epkNonce: number
+    setEpkNonce: (epkNonce: number) => void
+}
+
+const ActionDetail = ({
+    showBorder,
+    showHelp,
+    showRep,
+    maxRep,
+    defaultRep,
+    hasRep,
+    showoffRep,
+    setShowoffRep,
+    allEpks,
+    useSubsidy,
+    chooseToUseSubsidy,
+    chooseToUsePersona,
+    epkNonce,
+    setEpkNonce,
+}: Props) => {
+    const onRepChange = (event: any) => {
+        setShowoffRep(+event.target.value)
+    }
+
+    return (
+        <div className="action-detail">
+            <div className="choose-from">
+                <div className="choices">
+                    <div
+                        className={useSubsidy ? 'choice chosen' : 'choice'}
+                        onClick={chooseToUseSubsidy}
+                    >
+                        Rep-Handout
+                    </div>
+                    <div
+                        className={useSubsidy ? 'choice' : 'choice chosen'}
+                        onClick={chooseToUsePersona}
+                    >
+                        Personas
+                    </div>
+                </div>
+                {showHelp && (
+                    <div className="help">
+                        <HelpWidget type={InfoType.subsidy} />
+                    </div>
+                )}
+            </div>
+            {hasRep > defaultRep ? (
+                <div
+                    className={
+                        showBorder ? 'info-detail info-border' : 'info-detail'
+                    }
+                >
+                    {useSubsidy ? (
```

Comment:
You’re right, I’ll modify it.
Make it as an independent component is because both vote box and writing field use similar UI, so it’s better to make the similar feature as an component, in my opinikn